modules/test-utils/src/generate-layer-tests.js

```diff
@@ -89,7 +89,9 @@ export function generateLayerTests({
     onAfterUpdate(params);
     // Default assert
     if (params.layer.isComposite) {
-      assert(params.subLayers.length, 'Layer should have sublayers');
+      if (Object.keys(params.layer.props).length) {
```

Comment:
Why not just check if `props.data` is empty? This cannot be reliable.

Reply:
Because it is uncertain that no sublayer will be generated when there is no data.

Reply:
What is the use case you're fixing for? Can you add description to the PR summary?

Reply:
If you check `GeoJsonLayer`, it does not render any sub layers if `data` is empty. This test utility must work with the most common denominator. If a layer does not require `data` to render, you should add the additional check to its own tests.